src/main/cpp/src/parse_uri.cu

```diff
@@ -951,36 +957,105 @@ std::unique_ptr<column> parse_uri(strings_column_view const& input,
                              std::move(null_mask));
 }
 
+std::unique_ptr<column> parse_uri_ansi(strings_column_view const& input,
+                                       URI_chunks chunk,
+                                       std::optional<strings_column_view const> query_match,
+                                       bool ansi_mode,
+                                       rmm::cuda_stream_view stream,
+                                       rmm::device_async_resource_ref mr)
+{
+  // Get the result first
+  auto result = parse_uri(input, chunk, query_match, stream, mr);
+
+  // In ANSI mode, check for completely invalid URIs
+  if (ansi_mode && input.size() > 0) {
+    auto const d_strings = column_device_view::create(input.parent(), stream);
+
+    // Create boolean vector: true for valid URIs, false for completely invalid URIs
+    auto validity_flags = rmm::device_uvector<bool>(input.size(), stream);
+
+    thrust::transform(
+      rmm::exec_policy_nosync(stream),
+      thrust::counting_iterator<cudf::size_type>(0),
+      thrust::counting_iterator<cudf::size_type>(input.size()),
+      validity_flags.begin(),
+      cuda::proclaim_return_type<bool>([input = *d_strings] __device__(cudf::size_type row_idx) {
+        // Input nulls remain null
+        if (input.is_null(row_idx)) return false;
+
+        auto const in_string = input.element<string_view>(row_idx);
+        auto const uri =
+          validate_uri(in_string.data(), in_string.size_bytes(), cuda::std::nullopt, row_idx);
+
+        return uri.valid != 0;
+      }));
+
+    auto [validation_mask, null_count] = cudf::detail::valid_if(
+      validity_flags.begin(), validity_flags.end(), cuda::std::identity{}, stream, mr);
+
+    // Create validation column for throw_row_error_if_any
+    auto validation_column = cudf::make_numeric_column(cudf::data_type{cudf::type_id::INT8},
+                                                       input.size(),
+                                                       std::move(validation_mask),
+                                                       null_count,
+                                                       stream,
+                                                       mr);
```

Comment:
This makes a new column which allocates a memory buffer that is unused. Let's reuse the validity flags.
```
std::make_unique<column>(std::move(validity_flags), null_count > 0 ? std::move(validation_mask) : rmm::device_buffer{0, stream}, null_count);
```